<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Derivatives</class>
 <widget class="QDialog" name="Derivatives">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1203</width>
    <height>966</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
    <horstretch>0</horstretch>
    <verstretch>10</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Derivatives Lab</string>
  </property>
  <widget class="QTabWidget" name="tabWidget">
   <property name="geometry">
    <rect>
     <x>40</x>
     <y>160</y>
     <width>1141</width>
     <height>731</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Tahoma</family>
     <pointsize>10</pointsize>
    </font>
   </property>
   <property name="toolTip">
    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;br/&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
   <property name="autoFillBackground">
    <bool>false</bool>
   </property>
   <property name="currentIndex">
    <number>0</number>
   </property>
   <widget class="QWidget" name="tab_3">
    <attribute name="title">
     <string>Brownian Motion</string>
    </attribute>
    <widget class="MPL_WIDGET" name="oMplCanvas" native="true">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>851</width>
       <height>671</height>
      </rect>
     </property>
     <property name="mouseTracking">
      <bool>true</bool>
     </property>
     <property name="tabletTracking">
      <bool>true</bool>
     </property>
     <property name="autoFillBackground">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QTableWidget" name="tableWidget">
     <property name="geometry">
      <rect>
       <x>880</x>
       <y>360</y>
       <width>0</width>
       <height>321</height>
      </rect>
     </property>
     <property name="autoFillBackground">
      <bool>false</bool>
     </property>
     <property name="showGrid">
      <bool>false</bool>
     </property>
     <property name="cornerButtonEnabled">
      <bool>false</bool>
     </property>
    </widget>
    <widget class="MPL_WIDGET" name="oMplCanvas2" native="true">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>860</x>
       <y>10</y>
       <width>261</width>
       <height>411</height>
      </rect>
     </property>
     <property name="mouseTracking">
      <bool>true</bool>
     </property>
     <property name="tabletTracking">
      <bool>true</bool>
     </property>
     <property name="autoFillBackground">
      <bool>false</bool>
     </property>
    </widget>
    <widget class="QLCDNumber" name="lcdNum_prob">
     <property name="geometry">
      <rect>
       <x>1070</x>
       <y>750</y>
       <width>40</width>
       <height>23</height>
      </rect>
     </property>
     <property name="autoFillBackground">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_Copy2Clipboard">
     <property name="geometry">
      <rect>
       <x>980</x>
       <y>450</y>
       <width>131</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>9</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Copy to Clipboard</string>
     </property>
    </widget>
    <widget class="QGroupBox" name="groupBox">
     <property name="geometry">
      <rect>
       <x>900</x>
       <y>560</y>
       <width>211</width>
       <height>111</height>
      </rect>
     </property>
     <property name="title">
      <string/>
     </property>
     <widget class="QTextEdit" name="txt_yLim">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>60</y>
        <width>104</width>
        <height>31</height>
       </rect>
      </property>
      <property name="html">
       <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Tahoma'; font-size:10pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;300&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
      </property>
     </widget>
     <widget class="QCheckBox" name="checkBox_yLim">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>10</y>
        <width>181</width>
        <height>41</height>
       </rect>
      </property>
      <property name="text">
       <string>Fix Y-axis upper limit</string>
      </property>
     </widget>
    </widget>
   </widget>
  </widget>
  <widget class="QLabel" name="label">
   <property name="geometry">
    <rect>
     <x>20</x>
     <y>10</y>
     <width>1151</width>
     <height>51</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Microsoft JhengHei UI</family>
     <weight>75</weight>
     <bold>true</bold>
    </font>
   </property>
   <property name="text">
    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:14pt; color:#aa0000;&quot;&gt;TaiwanTech Derivatives Lab&lt;/span&gt;&lt;span style=&quot; font-size:14pt; color:#aa0000; vertical-align:super;&quot;&gt;TM&lt;/span&gt;&lt;span style=&quot; font-size:14pt; color:#313194;&quot;&gt;    Option Pricing and Strategies 選擇權定價與策略 Module 2&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
  </widget>
  <widget class="QDialogButtonBox" name="buttonBox">
   <property name="geometry">
    <rect>
     <x>1040</x>
     <y>920</y>
     <width>121</width>
     <height>41</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Tahoma</family>
     <pointsize>10</pointsize>
    </font>
   </property>
   <property name="orientation">
    <enum>Qt::Horizontal</enum>
   </property>
   <property name="standardButtons">
    <set>QDialogButtonBox::Close</set>
   </property>
  </widget>
  <widget class="QSlider" name="slider_t2">
   <property name="geometry">
    <rect>
     <x>130</x>
     <y>930</y>
     <width>471</width>
     <height>22</height>
    </rect>
   </property>
   <property name="maximum">
    <number>100</number>
   </property>
   <property name="singleStep">
    <number>5</number>
   </property>
   <property name="pageStep">
    <number>5</number>
   </property>
   <property name="value">
    <number>50</number>
   </property>
   <property name="orientation">
    <enum>Qt::Horizontal</enum>
   </property>
  </widget>
  <widget class="QSlider" name="slider_t1">
   <property name="geometry">
    <rect>
     <x>130</x>
     <y>900</y>
     <width>471</width>
     <height>22</height>
    </rect>
   </property>
   <property name="maximum">
    <number>100</number>
   </property>
   <property name="singleStep">
    <number>5</number>
   </property>
   <property name="pageStep">
    <number>5</number>
   </property>
   <property name="value">
    <number>25</number>
   </property>
   <property name="orientation">
    <enum>Qt::Horizontal</enum>
   </property>
   <property name="tickPosition">
    <enum>QSlider::NoTicks</enum>
   </property>
  </widget>
  <widget class="QLabel" name="label_9">
   <property name="geometry">
    <rect>
     <x>20</x>
     <y>900</y>
     <width>61</width>
     <height>21</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>10</pointsize>
    </font>
   </property>
   <property name="text">
    <string>Time 1</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_11">
   <property name="geometry">
    <rect>
     <x>20</x>
     <y>930</y>
     <width>61</width>
     <height>21</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>10</pointsize>
    </font>
   </property>
   <property name="text">
    <string>Time 2</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_t1">
   <property name="geometry">
    <rect>
     <x>610</x>
     <y>890</y>
     <width>114</width>
     <height>33</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>10</pointsize>
    </font>
   </property>
   <property name="text">
    <string>label_t1</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_t2">
   <property name="geometry">
    <rect>
     <x>610</x>
     <y>920</y>
     <width>114</width>
     <height>33</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>10</pointsize>
    </font>
   </property>
   <property name="text">
    <string>label_t2</string>
   </property>
  </widget>
  <widget class="QGroupBox" name="groupBox_2">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>70</y>
     <width>1001</width>
     <height>81</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Tahoma</family>
     <pointsize>10</pointsize>
    </font>
   </property>
   <property name="title">
    <string>Parameters</string>
   </property>
   <widget class="QSlider" name="oSlider_mu">
    <property name="geometry">
     <rect>
      <x>110</x>
      <y>50</y>
      <width>81</width>
      <height>16</height>
     </rect>
    </property>
    <property name="minimum">
     <number>-50</number>
    </property>
    <property name="maximum">
     <number>50</number>
    </property>
    <property name="pageStep">
     <number>1</number>
    </property>
    <property name="value">
     <number>5</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
    <property name="tickInterval">
     <number>5</number>
    </property>
   </widget>
   <widget class="QLabel" name="label_mu">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>22</y>
      <width>81</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>α = </string>
    </property>
   </widget>
   <widget class="QComboBox" name="comboBox_SimSize">
    <property name="geometry">
     <rect>
      <x>870</x>
      <y>50</y>
      <width>101</width>
      <height>20</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>100000</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>200000</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>500000</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>1000000</string>
     </property>
    </item>
   </widget>
   <widget class="QLabel" name="label_6">
    <property name="geometry">
     <rect>
      <x>860</x>
      <y>20</y>
      <width>121</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Simulation size</string>
    </property>
   </widget>
   <widget class="QSlider" name="oSlider_sigma">
    <property name="geometry">
     <rect>
      <x>210</x>
      <y>50</y>
      <width>91</width>
      <height>16</height>
     </rect>
    </property>
    <property name="minimum">
     <number>1</number>
    </property>
    <property name="maximum">
     <number>6</number>
    </property>
    <property name="pageStep">
     <number>1</number>
    </property>
    <property name="value">
     <number>2</number>
    </property>
    <property name="sliderPosition">
     <number>2</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
    <property name="tickInterval">
     <number>5</number>
    </property>
   </widget>
   <widget class="QLabel" name="label_sigma">
    <property name="geometry">
     <rect>
      <x>220</x>
      <y>13</y>
      <width>91</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>9</pointsize>
     </font>
    </property>
    <property name="text">
     <string>σ =</string>
    </property>
   </widget>
   <widget class="QSlider" name="oSlider_PathsShown">
    <property name="geometry">
     <rect>
      <x>760</x>
      <y>50</y>
      <width>75</width>
      <height>16</height>
     </rect>
    </property>
    <property name="mouseTracking">
     <bool>true</bool>
    </property>
    <property name="toolTip">
     <string>No. of paths to be shown</string>
    </property>
    <property name="minimum">
     <number>10</number>
    </property>
    <property name="maximum">
     <number>100</number>
    </property>
    <property name="singleStep">
     <number>10</number>
    </property>
    <property name="pageStep">
     <number>10</number>
    </property>
    <property name="value">
     <number>30</number>
    </property>
    <property name="sliderPosition">
     <number>30</number>
    </property>
    <property name="tracking">
     <bool>false</bool>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
    <property name="tickPosition">
     <enum>QSlider::NoTicks</enum>
    </property>
    <property name="tickInterval">
     <number>10</number>
    </property>
   </widget>
   <widget class="QSlider" name="oSlider_TimeStep">
    <property name="enabled">
     <bool>false</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>670</x>
      <y>50</y>
      <width>75</width>
      <height>16</height>
     </rect>
    </property>
    <property name="minimum">
     <number>5</number>
    </property>
    <property name="maximum">
     <number>25</number>
    </property>
    <property name="singleStep">
     <number>5</number>
    </property>
    <property name="value">
     <number>25</number>
    </property>
    <property name="sliderPosition">
     <number>25</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
    <property name="tickInterval">
     <number>5</number>
    </property>
   </widget>
   <widget class="QLabel" name="label_P">
    <property name="geometry">
     <rect>
      <x>763</x>
      <y>24</y>
      <width>81</width>
      <height>21</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Tahoma</family>
      <pointsize>9</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Paths = </string>
    </property>
   </widget>
   <widget class="QLabel" name="label_T">
    <property name="geometry">
     <rect>
      <x>670</x>
      <y>26</y>
      <width>81</width>
      <height>20</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>9</pointsize>
     </font>
    </property>
    <property name="text">
     <string>T= 1</string>
    </property>
   </widget>
   <widget class="QSlider" name="slider_S0">
    <property name="geometry">
     <rect>
      <x>550</x>
      <y>50</y>
      <width>91</width>
      <height>16</height>
     </rect>
    </property>
    <property name="minimum">
     <number>50</number>
    </property>
    <property name="maximum">
     <number>200</number>
    </property>
    <property name="singleStep">
     <number>5</number>
    </property>
    <property name="value">
     <number>100</number>
    </property>
    <property name="sliderPosition">
     <number>100</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
    <property name="tickInterval">
     <number>5</number>
    </property>
   </widget>
   <widget class="QLabel" name="label_S0">
    <property name="geometry">
     <rect>
      <x>550</x>
      <y>10</y>
      <width>101</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>9</pointsize>
     </font>
    </property>
    <property name="text">
     <string>y0 =</string>
    </property>
   </widget>
   <widget class="QSlider" name="slider_K">
    <property name="geometry">
     <rect>
      <x>440</x>
      <y>50</y>
      <width>91</width>
      <height>16</height>
     </rect>
    </property>
    <property name="minimum">
     <number>50</number>
    </property>
    <property name="maximum">
     <number>200</number>
    </property>
    <property name="singleStep">
     <number>1</number>
    </property>
    <property name="pageStep">
     <number>5</number>
    </property>
    <property name="value">
     <number>100</number>
    </property>
    <property name="sliderPosition">
     <number>100</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
    <property name="tickInterval">
     <number>5</number>
    </property>
   </widget>
   <widget class="QLabel" name="label_K">
    <property name="geometry">
     <rect>
      <x>440</x>
      <y>15</y>
      <width>101</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>9</pointsize>
     </font>
    </property>
    <property name="text">
     <string>K = </string>
    </property>
   </widget>
   <widget class="QRadioButton" name="radioRW">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>20</y>
      <width>61</width>
      <height>37</height>
     </rect>
    </property>
    <property name="text">
     <string>BM</string>
    </property>
   </widget>
   <widget class="QRadioButton" name="radioGRW">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>50</y>
      <width>71</width>
      <height>37</height>
     </rect>
    </property>
    <property name="text">
     <string>GBM</string>
    </property>
    <property name="checked">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QSlider" name="oSlider_deltaT">
    <property name="geometry">
     <rect>
      <x>320</x>
      <y>50</y>
      <width>101</width>
      <height>16</height>
     </rect>
    </property>
    <property name="minimum">
     <number>0</number>
    </property>
    <property name="maximum">
     <number>3</number>
    </property>
    <property name="pageStep">
     <number>1</number>
    </property>
    <property name="value">
     <number>2</number>
    </property>
    <property name="sliderPosition">
     <number>2</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
    <property name="tickInterval">
     <number>5</number>
    </property>
   </widget>
   <widget class="QLabel" name="label_deltaT">
    <property name="geometry">
     <rect>
      <x>330</x>
      <y>10</y>
      <width>91</width>
      <height>40</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>9</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Δt =</string>
    </property>
   </widget>
  </widget>
  <widget class="QPushButton" name="pushButton_ShowRandomWalk">
   <property name="geometry">
    <rect>
     <x>1050</x>
     <y>90</y>
     <width>111</width>
     <height>41</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Tahoma</family>
     <pointsize>10</pointsize>
    </font>
   </property>
   <property name="toolTip">
    <string>Simulate!</string>
   </property>
   <property name="text">
    <string>Simulate </string>
   </property>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>MPL_WIDGET</class>
   <extends>QWidget</extends>
   <header location="global">mplwidget</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>Derivatives</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>390</x>
     <y>380</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>Derivatives</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>390</x>
     <y>380</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
